packages/python-packages/doc-warden/warden/index_packages.py

```diff
@@ -0,0 +1,350 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from __future__ import print_function
+
+from .warden_common import check_match, walk_directory_for_pattern, get_omitted_files, get_java_package_roots, get_net_packages, get_python_package_roots, get_js_package_roots, find_alongside_file, find_below_file, parse_pom, is_java_pom_package_pom, find_above_file
+from .PackageInfo import PackageInfo
+import json
+import os
+import ast
+from jinja2 import Template
+import xml.etree.ElementTree as ET
+import textwrap
+import re
+import fnmatch
+
+# python 3 transitioned StringIO to be part of `io` module. 
+# python 2 needs the old version however
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+# chose to go with in-line templates. reasoning here is that the whitespace is a bit more important for markdown
+# given that all the content fitting on a single line is important, leveraging the format string method makes for a 
+# much simpler to maintain template
+PKGID_COL = ' [`{{ pkg.package_id }}`]( {{ pkg.relative_package_location }} )'
+RM_COL = ' {% if len(pkg.relative_readme_location) > 0 %}[Readme]({{ pkg.relative_readme_location }}){% else %} N/A {% endif %} '
+CL_COL = ' {% if len(pkg.relative_changelog_location) > 0 %}[Changelog]({{ pkg.relative_changelog_location }}){% else %} N/A {% endif %} '
+GROUPID_COL = ' `{{ pkg.repository_args[0] }}` '
+REPO_COL = ' `[Repo Link]( {{ pkg.relative_package_location }}` )'
+PUBLISH_COL = ' {% if pkg.test_url(config) %}[{{ pkg.get_repository_link_text(config) }}]( {{ pkg.get_formatted_repository_url(config) }} ){% else %} N/A {% endif %} '
+COLUMN_LOOP = '{% for pkg in packages %}'
+COLUMN_TEMPLATE = '{0}|{1}|{2}|{3}|{4}|'.format(COLUMN_LOOP, PKGID_COL, RM_COL, CL_COL, PUBLISH_COL)
+JAVA_COLUMN_TEMPLATE = '{0}|{1}|{2}|{3}|{4}|'.format(COLUMN_LOOP, PKGID_COL, GROUPID_COL, RM_COL, PUBLISH_COL)
+
+OUTPUT_HEADER = """
+# Package Index - {{ title }}
+
+| Package Id     | Readme    | Changelog                 | Published Url       |
+|----------------|-----------|---------------------------|---------------------|
+"""
+
+JAVA_OUTPUT_HEADER = """
+# Package Index - {{ title }}
+
+| Package Id     | GroupId   | Readme    | Published Url       |
```

Comment:
If we are using the java language this would be artifact id instead of package id.

Reply:
You're absolutely correct. Thanks for the catch.